### DISCUSSION TITLE
An error is reported when the length of int type in response_model exceeds the specified length

### DISCUSSION THREAD

**Discussion by jingyifun** [fastapi/fastapi#5989] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from pydantic import BaseConfig, BaseModel
from pydantic.fields import Field


class InDBBase(BaseModel):
    id: int = Field(title='primary key ID', description='primary key ID')

    class Config(BaseConfig):
        orm_mode = True
```

### Description

An error is displayed when the length of int type in response_model exceeds 16 bits

example
id: int = 109037141215586310
The final display is: id = 109037141215586300

The length of the data in the example is 18 bits

### Operating System

macOS

### FastAPI Version

0.91.0

### Python Version

3.11.1

**Comment by aayushchhabra1999** [comment]
I was unable to reproduce this by doing something like:
![image](https://user-images.githubusercontent.com/34938734/218255400-33d252fb-3c8b-4eaa-9872-4c1124f0c668.png)

Is the error coming from FastAPI or the database? 
Generally, the limit on Integer datatype for relational databases is 16 bits. Beyond that, you should be using BigInt.
Correct me if I misunderstood the problem.

**Reply by jingyifun** [reply]
First of all, thank you very much, I will explain the problem in the form of complete code
<img width="821" alt="image" src="https://user-images.githubusercontent.com/119286254/218360918-17684248-1e5a-4dc3-974f-2a8629d731ae.png">
<img width="1217" alt="image" src="https://user-images.githubusercontent.com/119286254/218361055-feee0b1f-c67f-48f0-9f31-c45ae8c1a11f.png">
Help to find the problem together, thank you

**Reply by jingyifun** [reply, marked as answer]
<img width="1415" alt="image" src="https://user-images.githubusercontent.com/119286254/218419241-d2cfa71a-4953-4da5-bcba-ff345b7b2edc.png">
The final reason is: the length of the id returned by the backend exceeds the maximum length of javascript number (16 bits), and the value overflows. Here is just the error message displayed.
solution:
1. The int type data in the backend should not exceed 16 bits;
2. In the returned model type, change id: int to id: str
I don't know if there is any good solution! Thanks

**Comment by aayushchhabra1999** [comment]
I found a related issue here:
https://github.com/tiangolo/sqlmodel/issues/191

It suggests something like:
 `id: int = Field(default_factory=next_val, sa_column=Column(BigInteger(), primary_key=True, autoincrement=False))`
